### PR TITLE
refactor(cb2-7174): add unique ids to components

### PR DESCRIPTION
**Pull request by owen-corrigan-bjss**
## CB2 -7174

https://dvsa.atlassian.net/browse/CB2-7174

adding ID's to fields, links and buttons without ID's, and tried to remove any duplicate ID's I have come across

### CODE REVIEW

**Review by gjulien-bjss — COMMENTED** [review]
You've got some failing tests

File: src/app/features/tech-record/components/tech-record-history/tech-record-history.component.html [hunk 1]
```diff
@@ -13,11 +13,13 @@
...
-        <td class="govuk-table__cell word-break">{{ techRecord.reasonForCreation }}</td>
-        <td class="govuk-table__cell word-break">{{ techRecord.createdByName }}</td>
-        <td class="govuk-table__cell">{{ techRecord.createdAt | date: 'dd/MM/yyyy HH:mm' }}</td>
+      <tr class="govuk-table__row" *ngFor="let techRecord of techRecords; trackBy: trackByFn; let i = index">
+        <th scope="row" class="govuk-table__header" [id]="[techRecord.statusCode] + '-status-header-' + [i]">{{ techRecord.statusCode }}</th>
+        <td class="govuk-table__cell word-break" [id]="[techRecord.statusCode] + '-reason-' + [i]">
+          {{ techRecord.reasonForCreation }}
+        </td>
+        <td class="govuk-table__cell word-break" [id]="[techRecord.statusCode] + '-created-by-' + [i]">{{ techRecord.createdByName }}</td>
+        <td class="govuk-table__cell" [id]="[techRecord.statusCode] + '-created-at-' + [i]">{{ techRecord.createdAt | date: 'dd/MM/yyyy HH:mm' }}</td>
```
Comment (gjulien-bjss) [inline src/app/features/tech-record/components/tech-record-history/tech-record-history.component.html:22]:
```suggestion
        <th scope="row" class="govuk-table__header" [id]="techRecord.statusCode + '-status-header-' + i">{{ techRecord.statusCode }}</th>
        <td class="govuk-table__cell word-break" [id]="techRecord.statusCode + '-reason-' + i">
          {{ techRecord.reasonForCreation }}
        </td>
        <td class="govuk-table__cell word-break" [id]="techRecord.statusCode + '-created-by-' + i">{{ techRecord.createdByName }}</td>
        <td class="govuk-table__cell" [id]="[techRecord.statusCode] + '-created-at-' + i">{{ techRecord.createdAt | date: 'dd/MM/yyyy HH:mm' }}</td>
```
Not sure why you would need brackets, unless I'm missing something

File: src/app/features/tech-record/components/test-record-summary/test-record-summary.component.html [hunk 2]
```diff
@@ -16,10 +16,12 @@ <h3 class="govuk-heading-s">No test records found.</h3>
...
-      <tr *ngFor="let test of paginatedTestFields; trackBy: trackByFn">
-        <th scope="row" class="govuk-table__header">{{ test.testTypeStartTimestamp | date: 'dd/MM/yyyy HH:mm' }}</th>
-        <td class="govuk-table__cell">{{ test.testTypeName }}</td>
-        <td class="govuk-table__cell">{{ test.testResult }}</td>
+      <tr *ngFor="let test of paginatedTestFields; trackBy: trackByFn; let i = index">
+        <th scope="row" class="govuk-table__header" [id]="'test-record-summary-date-' + [i]">
+          {{ test.testTypeStartTimestamp | date: 'dd/MM/yyyy HH:mm' }}
+        </th>
+        <td class="govuk-table__cell" [id]="'test-record-summary-name-' + [i]">{{ test.testTypeName }}</td>
+        <td class="govuk-table__cell" [id]="'test-record-summary-result-' + [i]">{{ test.testResult }}</td>
```
Comment (gjulien-bjss) [inline src/app/features/tech-record/components/test-record-summary/test-record-summary.component.html:24]: Same comment about the brackets

File: src/app/forms/components/dynamic-form-group/dynamic-form-group.component.html [hunk 3]
```diff
@@ -13,7 +13,13 @@
     <ng-container *ngSwitchCase="formNodeViewTypes.SUBHEADING">
       <div class="subheading-container">
         <h2 class="govuk-heading-m">{{ form.meta.label }}</h2>
-        <a *ngIf="form.meta.subHeadingLink" class="govuk-link" [routerLink]="form.meta.subHeadingLink.url">{{ form.meta.subHeadingLink.label }}</a>
+        <a
+          *ngIf="form.meta.subHeadingLink"
+          class="govuk-link"
+          [id]="'select-' + [form.meta.subHeadingLink.url] + '-link'"
```
Comment (gjulien-bjss) [inline src/app/forms/components/dynamic-form-group/dynamic-form-group.component.html:19]: brackets

File: src/app/forms/custom-sections/psv-brakes/psv-brakes.component.html [hunk 4]
```diff
@@ -11,7 +11,7 @@
 ></app-switchable-input>
 
 <ng-template #view [formGroup]="brakesForm">
-  <app-read-only formControlName="brakeCode" label="Brake code"></app-read-only>
+  <app-read-only formControlName="brakeCode" name="brake-code" label="Brake code"></app-read-only>
```
Comment (mcadenas-bjss) [inline src/app/forms/custom-sections/psv-brakes/psv-brakes.component.html:14]: Make sure this isn't breaking anything to do with forms.

File: src/app/shared/components/number-plate/number-plate.component.html [hunk 5]
```diff
@@ -1,3 +1,3 @@
-<span class="govuk-number-plate" [ngClass]="{ secondary: isSecondary }">
+<span class="govuk-number-plate" [id]="[isSecondary ? 'previous-vrm-span' : 'current-vrm-span']" [ngClass]="{ secondary: isSecondary }">
```
Comment (gjulien-bjss) [inline src/app/shared/components/number-plate/number-plate.component.html:1]:
```suggestion
<span class="govuk-number-plate" [id]="isSecondary ? 'previous-vrm-span' : 'current-vrm-span'" [ngClass]="{ secondary: isSecondary }">
```

**Review by gjulien-bjss — DISMISSED** [review]

File: src/app/forms/custom-sections/weights/weights.component.html [hunk 6]
```diff
@@ -37,11 +37,21 @@
...
-            <ng-template [ngTemplateOutlet]="axleTdNumberInput" [ngTemplateOutletContext]="{ name: 'designWeight' }"></ng-template>
+            <ng-template
+              [ngTemplateOutlet]="axleTdNumberInput"
+              [ngTemplateOutletContext]="{ name: 'gbWeightAxel' + axle.value.axleNumber }"
+            ></ng-template>
+
+            <ng-template
+              *ngIf="!isPsv"
+              [ngTemplateOutlet]="axleTdNumberInput"
+              [ngTemplateOutletContext]="{ name: 'eecWeight' + +axle.value.axleNumber }"
```
[the remark this answers is not in the document]
Comment (tahayazici) [reply src/app/forms/custom-sections/weights/weights.component.html]: `+` in front of a variable would try to convert that to a `number`, which is not needed here indeed.

**Review by gjulien-bjss — APPROVED** [review]

**Review by mcadenas-bjss — APPROVED** [review]

**Pull request merged**